.github/workflows/build-and-test.yaml

```diff
@@ -2,8 +2,11 @@
 
 on:
   push:
+    branches:
```

Comment:
It might be beneficial to specify the types of pull_request events that should trigger the workflow. For example, 'synchronize' to run tests on updates to a PR's branch. This can be done by adding a 'types' key under 'pull_request'. [medium]